### ISSUE TITLE
Maximize button not working properly with auto-hide taskbar enabled

### ISSUE THREAD

**Issue by lllopo** [microsoft/vscode#81102]
Version: 1.39.0-insider (user setup)
Commit: cc481ab31b5827add5f99a770f2c860d0575df69
Date: 2019-09-18T07:02:14.165Z
Electron: 6.0.9
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

Steps to Reproduce:

1. Start VS code in normal window state
2. Maximize it through the maximize button

Expected result - VS Code maximizes and the button changes to it's maximized glyph, user is able to restore to normal window through the same button
Actual result - VS Code maximizes visually, BUT the window actually just takes the screen size while remains draggable and the glyph remains the one for the normal state. The user is unable to restore normal state by clicking ti again.

Note : the maximize/restore by double clicking the title bar works normally.

Does this issue occur when all extensions are disabled?: Yes

**Comment by lllopo** [comment]
No, it is not #79902 :-)

**Comment by ABeeinSpace** [comment]
I see the expected behavior on Windows 10 ver. 18362. VS Code maximizes, the glyph changes, and when dragged the window properly restores down before moving.
I'm on VS Code 1.38.1, Electron 4.2.10 
Chrome 69
Node 10.11.0
V8: 6.9.427.31

**Comment by lllopo** [comment]
In addition - I said maximize/restore works properly through double-clicking the title bar, but this is not entirely true. When maximized, VS Code prevents the windows taskbar from showing up (it is set to autohide on my system).

**Comment by ABeeinSpace** [comment]
I was using the maximize button but didn't specify in my reply. My apologies. However, I'm still unable to reproduce the issue. I'm not at the PC I was using, and won't be for another 24 hrs. I can try to reproduce on another system but again that will take a few hours for me to be able to access it.

**Comment by lllopo** [comment]
> I was using the maximize button but didn't specify in my reply. My apologies. However, I'm still unable to reproduce the issue. I'm not at the PC I was using, and won't be for another 24 hrs. I can try to reproduce on another system but again that will take a few hours for me to be able to access it.

It is on insiders builds, appeared a few builds ago, so you can't reproduce it on stable branch anyway.

**Comment by sbatten** [comment]
if this is only reproducing in insiders, this is likely electron @deepak1556 @bpasero

**Comment by bpasero** [comment]
@sbatten can you try to reproduce too please? I cannot.

@lllopo did you configure the `window.titleBarStyle` setting?

**Comment by lllopo** [comment]
> @sbatten can you try to reproduce too please? I cannot.
> 
> @lllopo did you configure the `window.titleBarStyle` setting?

No, I don't have it (and it defaults to custom, I guess). When set to 'native', it works as intended, but that can be expected.

**Comment by bpasero** [comment]
@lllopo ok so to confirm, this only happens when `window.titleBarStyle: custom` is configured. What is your exact windows version (winver in cmd prompt)?

**Comment by lllopo** [comment]
> @lllopo ok so to confirm, this only happens when `window.titleBarStyle: custom` is configured. What is your exact windows version (winver in cmd prompt)?

Yes - insiders with custom title bar and Win version is the very latest : Win 10 1903 Build 18362.356

**Comment by sbatten** [comment]
I could not reproduce either

**Comment by lllopo** [comment]
[2019-09-25 11-25-53.zip](https://github.com/microsoft/vscode/files/3651424/2019-09-25.11-25-53.zip)

I tried to record a video of how my VS code behaves. Unfortunately it doesn't visualize the moments when I click with the mouse, but still I hope it would be helpful. Please, just assume when I go over the maximize button without shaking the cursor, I click it.

At the end I also recorded the misbehavior regarding the taskbar, but I guess it is related to the minimize/maximize problem overall. Anyway what it does is - start maximized and the windows taskbar is blocked from showing. After the sequance : doubleclick titlebar  to restore, doubleclick the titlebar to maximize again, click once the maximize button ... and the win taskbar gets unblocked and shows normally when I move the cursor to the bottom of the screen.

**Comment by lllopo** [comment]
> I can reproduce this as well. Same issues, very frustrating to work with.
> 
> https://streamable.com/mamyg
> 
> Win10, latest updates.

Yup - that's it.

**Comment by Paril** [comment]
I do confirm however it only happens with custom bar. Native one works great.

**Comment by sbatten** [comment]
@Paril @lllopo please test with today's insiders build. If you can still reproduce, please copy the text from Help > About here.

**Comment by Paril** [comment]
Seems to work now. The only issue I've ran into is that, rarely, when un-maximizing, all three of my monitors flicker on and off (sort of like when the GPU crashes, except there is no crash; like when desktop resolution changes kinda). It's random and not really reproducable though.

**Comment by sbatten** [comment]
Thanks for testing @Paril, sounds to me like another issue with electron 6, @bpasero

**Comment by lllopo** [comment]
@sbatten I also confirm - it is back to working as intended. Thanks for fixing it.

**Comment by lllopo** [comment]
I think this should be closed. It got fixed at my place, so it should go, if there are no other complaints.

**Comment by lllopo** [comment]
Ooookay, the issue returned as of : 
Version: 1.40.0-insider (user setup)
Commit: 72bc289d32ec4f867e73b7f1f2a22f7d614ae2bb
Date: 2019-10-11T05:44:05.506Z
Electron: 6.0.12
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by sbatten** [comment]
@lllopo makes sense as we've returned to electron 6 but are tracking this issue as part of that process

**Comment by eamodio** [comment]
Here is the related electron issue: https://github.com/electron/electron/issues/19934

**Comment by ednl** [comment]
Same problem here since I think the first 1.41 Insiders on latest Win 10 Pro: the window maximize button doesn't maximize the window, it just makes it as large as the screen (the window state and button appearance don't change) and the window doesn't restore when clicking again. To restore to smaller size, I have to resize by dragging a border, which shouldn't be possible in maximized state. Maximize/restore does work when doubleclicking the title bar but then the (auto-hide) Windows taskbar remains hidden unless I type the Win key. Version info:

Version: 1.41.0-insider (user setup)
Commit: 0d728c31ebdf03869d2687d9be0b017667c9ff37
Date: 2019-11-12T05:39:58.193Z
Electron: 6.1.4
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by deepak1556** [comment]
Just to confirm , does everyone seeing this issue have auto-hide taskbar enabled ? I am only able to repro under that condition. The videos from @Paril and @lllopo seems to suggest that.

**Comment by fireattack** [comment]
No. I don't have auto-hide taskbar.

**Comment by lllopo** [comment]
> Just to confirm , does everyone seeing this issue have auto-hide taskbar enabled ? I am only able to repro under that condition. The videos from @Paril and @lllopo seems to suggest that.

Yes, I get it only with the "autohide taskbar" enabled. When I disable it - it works fine.

**Comment by deepak1556** [comment]
@fireattack can you provide some exact repro steps and also if possible attach a video demonstrating it , so that I can repro it on my end.

Thanks for confirming @lllopo

**Comment by deepak1556** [comment]
https://github.com/electron/electron/pull/18321 possible cause of regression, looking into it,

**Comment by fireattack** [comment]
@deepak1556 

I investigated it further.

In my case, it was related to an instant messenger program, [QQ](https://im.qq.com/), the most popular IM in China.

The way it works is that you can have it sticked to the top or side of the screen (and always display on top). So you can reveal the main GUI by simply move the cursor to that part.

How it looks like when it is hides:

![image](https://user-images.githubusercontent.com/4130991/68748434-0beb8480-05c2-11ea-9d8d-4d49bb7587b2.png)

(notice the shadow)

How it looks when it reveals:

![image](https://user-images.githubusercontent.com/4130991/68748511-2cb3da00-05c2-11ea-94d0-f3804765c50f.png)

But this causes problem of the VS Code's maximizing function, like what others said:

>the window maximize button doesn't maximize the window, it just makes it as large as the screen (the window state and button appearance don't change) and the window doesn't restore when clicking again.

And this reminds me another program, Steam, which has a similar problem. In that case, you literally can't maximize at all (not even with double-click on title bar), it will always just make the window as big as the screen, sans 1px due to QQ.

I should stress, this is a regression. I've been using both VS Code and QQ for a very long time and there was no problem.

**Comment by deepak1556** [comment]
Thanks for the details @fireattack , yeah the root cause is same for this case.

**Comment by dcourtois** [comment]
Just had this issue after updating recently, glad it's already being investigated (I'm also auto hiding my task bar (on Windows 10) and it happens only with the title bar setting of VSCode set to "custom". Reverting to the native title bar fixes everything)

But I'd like to mention another problem that seems related : when vscode is maximized (either by double clicking the custom title bar or using the upper right corner icon) then my task bar will not show. I have to manually minimize my vscode to be able to show the task bar. e.g.:

- set your system's task bar in auto-hide and vscode in custom title bar
- launch vscode and maximize its window
- try to move the mouse cursor to where the task bar usually pops... nothing happens

If you use the native title bar, everything works, the task bar correctly pops when the mouse move over it.

**Comment by lllopo** [comment]
Also mentioned already. My temporary workaround for now is to click once one the maximize button. Although it’s still not working, it makes the taskbar appear agin for some unknown reason.

**Comment by tonygiang** [comment]
If I am reading this correctly, this entry is actually 2 different issues:
1. The maximize button on the upper right corner does not do anything
2. VSCode in maximized state does not allow autohide taskbar to trigger

**Comment by supposedly** [comment]
That's correct, but it was mentioned in #84342 that they're two manifestations of the same root issue.

**Comment by eeasoftware** [comment]
I am able to reproduce the error mentioned by dcourtois.  

It doesn't matter what other applications are running, when the focus is on vscode, the windows taskbar does not auto-pop-up. When I focus on another application in a different screen or a different application in the same screen (doesn't really matter), I can unhide/auto-hide the task bar.

For reference, this issue started for me after the latest update.
vscode: 1.40.1
os: windows 7

**Comment by deepak1556** [comment]
Thanks for the reports, the issue is being actively worked on , fix is already up on electron end https://github.com/electron/electron/pull/21164 and will soon be available in the next vscode stable recovery release later this week.

**Comment by lllopo** [comment]
It's over, latest insiders works fine again for me ! Hallelujah !

**Comment by deepak1556** [comment]
Fixed with https://github.com/microsoft/vscode/pull/85266

**Comment by klenium** [comment]
This is crazy. Electron had a bug (hey, what is testing?) that is already fixed by now. About 4 programs I use and depends on Electron released their program with the Electorn's bug (VS Code was updated today...), so now I can be pretty happy. This is amazing how many programs depend on other programs, so users will get these trivial bugs, one program brokes 5 other, and fixes for these 5 will come some months later. That's why we replaced stable GUI technologies like WPF?
